### ISSUE TITLE
CLI: allow Heap to be configured

### ISSUE THREAD

**Issue by alexanderdean** [snowplow/schema-guru#102]
To prevent Java Heap errors when processing huge files of JSONs (e.g. 7Gig).

As a workaround I am running:

    java -Xmx6g -classpath ./schema-guru-0.3.0 com.snowplowanalytics.schemaguru.Main schema 2015-08-12T08Z.log

**Comment by alexanderdean** [comment]
Other useful settings are:

```
-XX:MaxPermSize=128M -XX:+UseCompressedOops
```

as per http://stackoverflow.com/questions/5497259/how-to-fix-requested-array-size-exceeds-vm-limit-error-in-java

**Comment by chuwy** [comment]
Underlying problem is that we always load all JSONs straight into memory as `List[String]`. This can be avoided by using stream processing engines such as [fs2](https://github.com/functional-streams-for-scala/fs2). Same for NDJSON, we can split it into multiple chunks.

**Comment by alexanderdean** [comment]
Or should we be considering moving the local-mode inference to just use Spark local-mode - so reduce the code surface area of Schema Guru?

**Comment by chuwy** [comment]
No, I think it's better to leave spark job for clusters so far. Because we can benefit from `Stream[Json]` in Web UI as well.

**Comment by alexanderdean** [comment]
Okay, fair play!
